### DISCUSSION TITLE
spaCy PRFScore class adding score objects

### DISCUSSION THREAD

**Discussion by siddharth1199** [explosion/spaCy#6992] [Help: Coding & Implementations]
I am new to spaCy and have been running a model on a few annotations. I have a list of metrics for each of the annotations (a list of PRFScore objects) which look like this:
```
from spacy.scorer import PRFScore

l = [<spacy.scorer.PRFScore at 0x7f24757735f8>,
      <spacy.scorer.PRFScore at 0x7f24757736d8>,
      <spacy.scorer.PRFScore at 0x7f24a4490518>]

```

What I am trying to do is consolidate all the score objects into a single object, so that I can get a single precision/recall score for all the annotations. I understand the scorer class has a dunder [method](https://github.com/explosion/spaCy/blob/ca3f8386d789c0012965ad8cd3ea50bf55f11506/spacy/scorer.py#L37) for adding PRFScore objects, what I am not sure of is how to access it, and add all the 3 PRFScore objects into a single one.

What I have tried are the following steps:
1. `l[0]+l[1]` which gives the error `TypeError: unsupported operand type(s) for +: 'PRFScore' and 'PRFScore'`
2. `metrics[0]._PRFScore__add(metrics[1])` which gives the error `AttributeError: 'PRFScore' object has no attribute '_PRFScore__add'` (Accessing the dunder method using this Stack Overflow [link](https://stackoverflow.com/questions/44114560/how-to-access-double-underscore-variables-in-methods-added-to-a-class))

Any help on this will be greatly appreciated,
Thank you.

**Comment by adrianeboyd** [comment]
What should currently work in v3.0 is `prf1 += prf2`. It looks like the `__add__` implemention is incorrect, so `prf3 = prf1 + prf2` won't work correctly right now. You can also just access and add the `prf.tp/fp/fn` values by hand and set them in a newly created `PRFScore`. It's a really simple class, so you may just want to make a copy of the class with a custom name and fix the issues to make your task easier now?

If you're getting `TypeError: unsupported operand type(s) for +: 'PRFScore' and 'PRFScore'` you're probably using v2 instead of v3? The new `add/iadd` methods are only in v3.

**Reply by siddharth1199** [reply]
@adrianeboyd Thanks a lot, looks like if I want to use Prodigy with spCay 3 I'd need to sign up to the Prodigy nightly program. Will try that solution then!
